packages/xstate-fsm/src/types.ts

```diff
@@ -17,7 +17,7 @@ export namespace StateMachine {
 
   export interface ActionObject<TContext, TEvent> {
     type: string;
-    exec?: (context: TContext, event: TEvent) => void;
+    exec?: (context: TContext, event?: TEvent) => void;
```

Comment:
@davidkpiano are you ok with this change? this changes it for all actions, but at runtime, it affects only initial actions. It's unfortunate but I don't think there is an easy way to change this at TS level :s

Reply:
Might not be a good idea, in XState core we just provide a dummy initial action.

Reply:
> in XState core we just provide a dummy initial action.

Yeah, but we might need to remove this in v5 - if we want to be completely 100% SCXML-compatible.

Based on https://www.w3.org/TR/scxml/#SystemVariables :
> The SCXML Processor must not bind _event at initialization time until the first event is processed. Hence _event is unbound when the state machine starts up.

Reply:
The dummy event is more for TS; we can definitely remove it, but requiring devs to check for an undefined event will be a pretty bad developer experience, when it is almost always defined.

Reply:
I agree that this could deteriorate DX, this spreads quite badly.

It would be great though to get concrete examples of when this would become a pain - in places where a concrete event cannot be inferred a developer has to check for `.type` any way to do anything useful with contents of the event and given the fact that TS@3.7 has brought optional chaining to us `event?.type` doesn't look that bad in comparison to `event.type`. Places, where concrete event type can be inferred, wouldn't be affected.

There is also a general question here if we are willing to sacrifice SCXML compatibility in situations like this or is this a definite no go? This is a hard one, both ways have their pros & cons.

Reply:
Let's put a dummy event for now (re: offline discussion about SCXML compatibility)

Reply:
Ok, added a dummy event